pkg/helmdeployer/deployer.go

```diff
@@ -428,10 +428,23 @@ func (h *helm) Resources(bundleID, resourcesID string) (*deployer.Resources, err
 }
 
 func (h *helm) Delete(bundleID, releaseName string) error {
-	if releaseName != "" {
-		return h.deleteByRelease(bundleID, releaseName)
+	if releaseName == "" {
+		deployments, err := h.ListDeployments()
+		if err != nil {
+			return err
+		}
+		for _, deployment := range deployments {
+			if deployment.BundleID == bundleID {
+				releaseName = deployment.ReleaseName
+				break
+			}
+		}
+	}
+	if releaseName == "" {
+		// Never found anything to delete
+		return nil
 	}
```

Comment:
nit: we could probably move this into the scope above since this logic proceeds the for loop.
```go
if releaseName == "" {
    deployments, err := h.ListDeployments()
    // snip
    for _, deployment := range deployments {
    // snip
    }
    if releaseName == "" {
        return nil
    }
}